### ISSUE TITLE
report (&/or abort) on cdparanoia non-fatal errors

### ISSUE THREAD

**Issue by RecursiveForest** [JoeLametta/whipper#88]
If cdparanoia returns "!" (or any other code that should cause alarm), whipper should stop ripping or at minimum report an error.

**Comment by MerlijnWajer** [comment]
I'm repeating myself, but I have code for this. I will try to get it in soon, but if anyone wants to do it before I do, post here and I'll send my unclean patch.

**Comment by RecursiveForest** [comment]
Iff the patch you have contains the entirety of your support for it, I'd be happy to integrate it.

**Comment by MerlijnWajer** [comment]
Here are two patches. They will not outright apply, but the gist is in there:

(1) https://ptpb.pw/EbXI
(2) https://ptpb.pw/feyY

And here's a patch to parse cdrdao warnings:

(1) https://ptpb.pw/Iwb0

**Comment by MerlijnWajer** [comment]
Can we make this perhaps part of the next milestone after 1.0? I don't think we really -need- this now?

**Comment by tn5421** [comment]
My understanding of the rationale says that it probably is necessary for the 1.0 milestone, since acquiring the 'perfect rip' means we need to know if any kind of error occurs.

**Comment by 45054** [comment]
I'd also think that fixing / enhancing the error reporting should be part of the 1.0 milestone especially regarding the aforementioned bug #75 which would otherwise produce false logs.

**Comment by MerlijnWajer** [comment]
Fair enough. I've found that cdparanoia can spit out a lot of errors and still rip with 100% accuracy, after many retries and errors (and corrections). So they are not a completely accurate representation. But then again, maybe the ones in #75 are different from the errors/warnings I usually see.
